lib/portlayer/exec/exec.go

```diff
@@ -76,12 +86,27 @@ func Init(ctx context.Context, sess *session.Session, source extraconfig.DataSou
 			Config.ResourcePool = o.ResourcePool
 		case *object.ResourcePool:
 			Config.ResourcePool = o
+			rp := compute.NewResourcePool(ctx, sess, cr)
+			Config.Cluster, err = rp.GetCluster(ctx)
```

Comment:
Need to confirm this works in ROBO - not that VMgroup does but this is a general code path.

Reply:
I successfully ran the "Create a VCH without a VM group" test from this PR against an environment where DRS was disabled for the only cluster.

I also manually deployed a VCH in that same environment and used it to successfully create a container. I inspected the portlayer log to locate the `Beginning initialization of portlayer exec component` message (logged at the beginning of this method) and verified that the file did not contain the `could not get cluster from resource pool: ...` message (logged in the `err != nil` case below).